delocate/tools.py

```diff
@@ -148,6 +148,30 @@ def _run(
         ) from exc
 
 
+# Mach-O magic numbers. See:
+# https://github.com/apple-oss-distributions/xnu/blob/5c2921b07a2480ab43ec66f5b9e41cb872bc554f/EXTERNAL_HEADERS/mach-o/loader.h#L65
+# https://github.com/apple-oss-distributions/cctools/blob/658da8c66b4e184458f9c810deca9f6428a773a5/include/mach-o/fat.h#L48
+MACHO_MAGIC = (
+    0xFEEDFACE.to_bytes(4, "little"),  # MH_MAGIC
+    0xFEEDFACE.to_bytes(4, "big"),  # MH_MAGIC
+    0xFEEDFACF.to_bytes(4, "little"),  # MH_MAGIC_64
+    0xFEEDFACF.to_bytes(4, "big"),  # MH_MAGIC_64
+    0xCAFEBABE.to_bytes(4, "big"),  # FAT_MAGIC (always big-endian)
+    0xCAFEBABF.to_bytes(4, "big"),  # FAT_MAGIC_64 (always big-endian)
+)
+
+
+def _is_macho_file(filename: str) -> bool:
```

Comment:
Maybe add a short docstring to `_is_macho_file` to say what it actually does:
```suggestion
def _is_macho_file(filename: str) -> bool:
    """Return True if the file at `filename` begins with a Mach-O magic number."""
```